in method:bad, at line:31, inst:3 = new <Application,Ltestcases/CWE89_SQL_Injection/s04/CWE89_SQL_Injection__URLConnection_executeQuery_61b>@0
in method:bad, at line:31, inst:invokespecial < Application, Ltestcases/CWE89_SQL_Injection/s04/CWE89_SQL_Injection__URLConnection_executeQuery_61b, <init>()V > 3 @4 exception:4
in method:bad, at line:31, inst:6 = invokevirtual < Application, Ltestcases/CWE89_SQL_Injection/s04/CWE89_SQL_Injection__URLConnection_executeQuery_61b, badSource()Ljava/lang/String; > 3 @7 exception:5
in method:bad, at line:33, inst:9 = invokestatic < Application, Ltestcasesupport/IO, getDBConnection()Ljava/sql/Connection; > @18 exception:8
in method:bad, at line:34, inst:11 = invokeinterface < Application, Ljava/sql/Connection, createStatement()Ljava/sql/Statement; > 9 @23 exception:10
in method:bad, at line:35, inst:12 = new <Application,Ljava/lang/StringBuilder>@30
in method:bad, at line:39, inst:invokespecial < Application, Ljava/lang/StringBuilder, <init>()V > 12 @34 exception:13
in method:bad, at line:39, inst:16 = invokevirtual < Application, Ljava/lang/StringBuilder, append(Ljava/lang/String;)Ljava/lang/StringBuilder; > 12,14 @39 exception:15
in method:bad, at line:40, inst:18 = invokevirtual < Application, Ljava/lang/StringBuilder, append(Ljava/lang/String;)Ljava/lang/StringBuilder; > 16,6 @43 exception:17
in method:bad, at line:40, inst:21 = invokevirtual < Application, Ljava/lang/StringBuilder, append(Ljava/lang/String;)Ljava/lang/StringBuilder; > 18,19 @48 exception:20
in method:bad, at line:40, inst:23 = invokevirtual < Application, Ljava/lang/StringBuilder, toString()Ljava/lang/String; > 21 @51 exception:22
in method:bad, at line:40, inst:25 = invokeinterface < Application, Ljava/sql/Statement, executeQuery(Ljava/lang/String;)Ljava/sql/ResultSet; > 11,23 @54 exception:24
in method:badSource, at line:37, inst:4 = new <Application,Ljava/net/URL>@3
in method:badSource, at line:41, inst:invokespecial < Application, Ljava/net/URL, <init>(Ljava/lang/String;)V > 4,5 @9 exception:6
in method:badSource, at line:41, inst:8 = invokevirtual < Application, Ljava/net/URL, openConnection()Ljava/net/URLConnection; > 4 @12 exception:7
in method:badSource, at line:41, inst:10 = new <Application,Ljava/io/InputStreamReader>@21
in method:badSource, at line:41, inst:12 = invokevirtual < Application, Ljava/net/URLConnection, getInputStream()Ljava/io/InputStream; > 8 @26 exception:11
in method:badSource, at line:42, inst:invokespecial < Application, Ljava/io/InputStreamReader, <init>(Ljava/io/InputStream;Ljava/lang/String;)V > 10,12,13 @31 exception:14
in method:badSource, at line:43, inst:15 = new <Application,Ljava/io/BufferedReader>@36
in method:badSource, at line:47, inst:invokespecial < Application, Ljava/io/BufferedReader, <init>(Ljava/io/Reader;)V > 15,10 @42 exception:16
in method:badSource, at line:47, inst:18 = invokevirtual < Application, Ljava/io/BufferedReader, readLine()Ljava/lang/String; > 15 @47 exception:17
in method:badSource, at line:47, inst:conditional branch(eq, to iindex=32) 15,9
in method:badSource, at line:47, inst:invokevirtual < Application, Ljava/io/BufferedReader, close()V > 15 @56 exception:52
in method:badSource, at line:47, inst:54 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:47, inst:55 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:48, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 54,55,33,53 @74 exception:56
in method:badSource, at line:48, inst:conditional branch(eq, to iindex=44) 10,9
in method:badSource, at line:48, inst:invokevirtual < Application, Ljava/io/InputStreamReader, close()V > 10 @84 exception:58
in method:badSource, at line:53, inst:60 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:53, inst:61 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:53, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 60,61,39,59 @102 exception:62
in method:badSource, at line:64, inst:22 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:64, inst:23 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:66, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 22,23,24,19 @120 exception:25
in method:badSource, at line:72, inst:conditional branch(eq, to iindex=63) 20,9
in method:badSource, at line:69, inst:invokevirtual < Application, Ljava/io/BufferedReader, close()V > 20 @128 exception:41
in method:badSource, at line:71, inst:43 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:71, inst:44 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:71, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 43,44,33,42 @146 exception:45
in method:badSource, at line:71, inst:conditional branch(eq, to iindex=75) 21,9
in method:badSource, at line:71, inst:invokevirtual < Application, Ljava/io/InputStreamReader, close()V > 21 @156 exception:47
in method:badSource, at line:76, inst:49 = getstatic < Application, Ltestcasesupport/IO, logger, <Application,Ljava/util/logging/Logger> >
in method:badSource, at line:76, inst:50 = getstatic < Application, Ljava/util/logging/Level, WARNING, <Application,Ljava/util/logging/Level> >
in method:badSource, at line:76, inst:invokevirtual < Application, Ljava/util/logging/Logger, log(Ljava/util/logging/Level;Ljava/lang/String;Ljava/lang/Throwable;)V > 49,50,39,48 @174 exception:51
in method:badSource, at line:57, inst:return 63
